docs/guides/security-guide.md

```diff
@@ -1,141 +1,95 @@
+key and cert files, and client-side key and cert files. This results in a
+1-depth level certificate chain, which will suffice for validation and
+confirmation but may not provide sufficient security for production systems.
+It is highly recommended to use well-known CAs, and generate certificates at
+multiple depth levels in order to conform to higher security standards.
 
-All certificates are in PEM format.
+The reference files are available here:
+<https://github.com/stratum/stratum/tree/main/tools/tls>
 
-To generate the TLS certificates:
+### 2.1 Running in secure mode
 
-* Review the files `ca.conf` and `grpc-client.conf`, to verify that the configuration settings are as desired
-* Run the generate-certs.sh script with following command:
+#### 2.1.1 Requirements for gRPC server
 
-```bash
-COMMON_NAME=<IP> ./generate-certs.sh
-or
-COMMON_NAME=<FQDN> ./generate-certs.sh
-or
-COMMON_NAME=localhost ./generate-certs.sh
-```
-
-* Copy the generated ca.crt, stratum.crt, and stratum.key to the server
-  running infrap4d
-* Copy the generated ca.crt, client.crt, and client.key to the gRPC client
-  machine
-
-#### Certificate installation
-
-infrap4d will check for certificate in the following default location:
+The IPDK Networking Recipe uses a secure-by-default model. If you wish to
+open insecure ports, you must do so explicitly. It is strongly recommended
+that you use secure ports in production systems.
 
-> /usr/share/stratum/certs/
+infrap4d is launched with following gRPC ports secured via TLS certificates.
+The port numbers are:
 
-If alternate location is desired, the location will need to be specified
-during runtime with the following flags:
+- 9339 - an IANA-registered port for gNMI
+- 9559 - an IANA-registered port for P4RT
 
-```text
--ca_cert_file=[CA certificate file]
--server_cert_file=[Server certificate file]
--server_key_file=[Server private key file]
-```
+#### 2.1.2 Requirements for gRPC clients
 
-For example:
+Under default conditions, the gRPC clients will require the TLS certificates
+to establish communication with infrap4d. The clients will need
+to use the same ca.crt file and the client.key and client.crt files signed
+by the ca.crt (can copy the generated files from the server if client is not
+on the same system as server).
 
-```bash
-# Files present in /tmp/certs/ directory
-$IPDK_RECIPE/install/sbin/infrap4d \
-    -ca_cert_file=/tmp/certs/ca.crt \
-    -server_cert_file=/tmp/certs/stratum.crt \
-    -server_key_file=/tmp/certs/stratum.key
-```
+- P4RT client
 
-### Client certificate verification
+    The P4Runtime Control client will default to communicating in secure mode
+using port 9559. If certificates are not available, the P4RT client will attempt
+a connection using insecure client credentials as a fallback mechanism.
+Note that the communication will fail if infrap4d runs in secure mode.
+Both server and client must specify insecure mode for this to work.
```

Comment:
Change `Both server and client must specify insecure mode`
to `Server miust specify insecure mode`

since client does not have a flag to specify insecure mode

Reply:
I just realized that we have an option of insecure mode for gnmi client but not p4rt-ctl so we can't have a generic statement.